### PR TITLE
fix bug hooks and add fitur search

### PR DESCRIPTION
**Pull request by xrzky**

### CODE REVIEW

File: src/components/list.js [hunk 1]
```diff
@@ -1,41 +1,61 @@
...
-
-  const { pokemonData } = usePokemon();
+  const { filteredPokemon, searchQuery, handleSearchChange } = usePokemon();
 
   return (
     <>
+      <div className="mx-20 py-auto ">
+        <div className="relative">
+          <div className="absolute inset-y-0 flex items-center start-0 ps-3">
+            <img src="./search.svg" alt="search-icon" />
```
Comment (hanifwdyt) [inline src/components/list.js]: prefer use tag Image from nextjs
ref : https://nextjs.org/docs/pages/api-reference/components/image

File: src/components/list.js [hunk 2]
```diff
@@ -1,41 +1,61 @@
...
+            <input
+              type="search"
+              id="search"
+              className="block w-full p-2 text-sm text-black rounded-full ps-10 focus:outline-none"
+              placeholder="Search"
+              value={searchQuery}
+              onChange={handleSearchChange}
+              required
+            />
+            <img className="p-2 mx-4 bg-white rounded-full" src="./text_format.svg" alt="" />
```
Comment (hanifwdyt) [inline src/components/list.js]: also this one

File: src/components/list.js [hunk 3]
```diff
@@ -1,41 +1,61 @@
...
+            { filteredPokemon.map ((pokemon) => (
+              <div key={ pokemon.id }>
+                <Link href={{ pathname: `/pokemon/${ pokemon.id }` }}>
                   <div className="text-center no-underline bg-white rounded-lg shadow-2xl cursor-pointer w-36 h-36 drop-shadow">
                     <div className="px-2 py-2 text-right text-grey-text min-h-4">
-                      <p className="text-[12px] leading-3">#{String(index + 1).padStart(3, "0")}</p>
+                      <p className="text-[12px] leading-3">#{String( pokemon.id ).padStart(3, "0")}</p>
                     </div>
                     <div className="w-20 h-20 m-auto">
                       <Image className="w-full h-full"
```
Comment (hanifwdyt) [inline src/components/list.js]: if Image has width and height attribute, add classname w-full h-full is unnecesarry. read the documentation

File: src/components/list.js [hunk 4]
```diff
@@ -1,41 +1,61 @@
...
                     <div className="rounded-lg bg-grey-bg pt-7 pr-2 pb-3 pl-2 mt-[-18px] text-title-list">
-                      <p className="text-xs">{pokemon.name}</p>
+                      <p className="text-xs">{ pokemon.name }</p>
                     </div>
                   </div>
                 </Link>
               </div>
             ))}
+            { filteredPokemon.length === 0 && (
+              <div className="text-center">Pokemon tidak ditemukan</div>
```
Comment (hanifwdyt) [inline src/components/list.js:67]: dont forget add isLoading state

File: src/components/list.js (hunk 1 above)
Comment (xrzky) [reply src/components/list.js]: okay, i will change to Image from nextJs

File: src/components/list.js (hunk 2 above)
Comment (xrzky) [reply src/components/list.js]: okay, i will change to Image from nextJs

File: src/components/list.js (hunk 3 above)
Comment (xrzky) [reply src/components/list.js]: noted sir

File: src/components/list.js (hunk 4 above)
Comment (xrzky) [reply src/components/list.js:67]: noted sir

**Review by hanifwdyt — APPROVED** [review]

**Review by hanifwdyt — APPROVED** [review]

**Pull request merged**